block/manager.go

```diff
@@ -293,14 +315,21 @@ func (m *Manager) syncFromSettlement() error {
 		// TODO: separate between fresh rollapp and non-registered rollapp
 		return err
 	}
+
 	m.LastSubmittedHeight.Store(res.EndHeight)
-	err = m.syncToTargetHeight(res.EndHeight)
 	m.UpdateTargetHeight(res.EndHeight)
+
+	// get the latest finalized height to know from where to start validating
+	err = m.UpdateFinalizedHeight()
 	if err != nil {
 		return err
 	}
 
-	m.logger.Info("Synced.", "current height", m.State.Height(), "last submitted height", m.LastSubmittedHeight.Load())
+	// try to sync to last state update submitted on startup
+	m.triggerStateUpdateSyncing()
+	// try to validate all pending state updates on startup
+	m.triggerStateUpdateValidation()
```

Comment:
it needs to be called because it may happen that the node is already synced but there are pending state updates not validated when starting the node.  in that case, validation will only start after a new state update arrives.  moved it outside the function, though.

Reply:
replied beneath